src/button/__tests__/button.test.tsx

```diff
@@ -91,6 +91,104 @@ describe('Button Component', () => {
     });
   });
 
+  describe('disabled with reason', () => {
```

Comment:
I suggest adding another tests which checks that the buttons `onClick` is not fired when it's disabled with a reason.

Reply:
added